### ISSUE TITLE
Crash when no MAX_TREE_DEPTH is set

### ISSUE THREAD

**Issue by jmmcd** [jmmcd/PonyGE2#55]
This command crashes:

`python ponyge.py --error_metric rmse --fitness_function regression`

```
  File "PonyGE2/src/representation/grammar.py", line 543, in get_min_ramp_depth
    depths = range(self.min_path, max_tree_depth + 1)
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'
```

**Comment by dvpfagan** [comment]
So we should have a catch around it for rhh/tree stuff and then use the
ramp depth when doing init rather than max tree depth. Then throw an error
if no ramp depth is found between 0 and the max init tree depth.

**Comment by mikefenton** [comment]
The issue raised by James was actually a slight bug (fixed in commit 1343665693111642e7ea08c9d5b80fc4773869f3). However, this raises a number of bigger issues:

1. Our command line parser doesn't allow us to set "None" values for various parameters such as `max_tree_depth` as it expects an `int` for these.
2. Derivation and mapping needs to be massaged to allow for no depth limits.

The overall issue arose because I tried to address this first point by setting `MAX_TREE_DEPTH`, `MAX_GENOME_LENGTH`and `MAX_TREE_NODES` to `None` in the default `params` dictionary. This means that unless you specify these values directly in either a params file or from the command line, they will remain at `None`. We need to be explicit about this in the `README`! Another way to tackle this is to set any of these values to `0` from the command line. This will have the same effect as using `None`, as it will bypass the various checks needed.

For reference, `MAX_GENOME_LENGTH`and `MAX_TREE_NODES` are only checked in `utilities.representation.check_methods.check_ind`. `MAX_TREE_DEPTH`, however, is checked quite a lot, particularly in both derivation and mapping. I'm currently modifying these files to allow for `0` or `None` values for `MAX_TREE_DEPTH`.

**Comment by mikefenton** [comment]
Latest commit solves this issue.
